### ISSUE TITLE
feat: Multi-tenancy of Web UI

### ISSUE THREAD

**Issue by osterman** [mumoshu/variant2#17]
Importing from https://github.com/mumoshu/variant/issues/33

Add a Login page and a project-selection page in front of [#32](https://github.com/mumoshu/variant/issues/32). It should be deployed along with multiple instances of variant server [#31](https://github.com/mumoshu/variant/issues/31) (+ perhaps [#32](https://github.com/mumoshu/variant/issues/32)).

This may be a too big project that is unlikely to be implemented. But I'll write this just for sake of completeness .

**Comment by osterman** [comment]
Was talking about this with @johncblandii today. I understand if this could be too big of a project. 

Currently, the slack bot fills this role, but there's no ability to stream output for long-running commands. 

I imagine the UI could be similar to [shellhub](https://github.com/shellhub-io/shellhub) but instead of launching shells, it runs jobs and prompts with a form where options or parameters may be needed. Output from the command would be streamed to the console. 

## Use-case 
- Kiosk deployed as a container under kubernetes that exposes a specific set of commands to users with access. Secured behind Keycloak.

**Comment by johncblandii** [comment]
This would be sweet!

**Comment by mumoshu** [comment]
Yeah, this would be so nice to have! But I'm not that fast coder/good designer for Web UIs.

Could anyone donate this to Variant 2? 😃

**Comment by mumoshu** [comment]
@osterman Btw Keycloak in this picture provides authentication, but not authorization.

Probably we need an authorization system and RBAC functionality built-in to make this practically useful?

**Comment by mumoshu** [comment]
Where's the silver bullet for building an enterprise-grade Web UI today? :)

**Comment by osterman** [comment]
> Where's/What's the silver bullet for building an enterprise-grade Web UI today? :)

Heh, I'm probably more out of date than you on this. I really wish I knew other than I'm constantly impressed by how many nice UIs there are these days (long gone are the day of the unbearable nagios-grade UIs). 

Let me pull in @zloeber who has an answer for everything. =)

**Comment by Gowiem** [comment]
@osterman @mumoshu Saw this while browsing the #variant channel. On a project I manage we went with [React Admin](https://github.com/marmelab/react-admin), which I think would be a good fit to bootstrap this use-case.
